admin/app/routes/authenticated/tools.js

```diff
@@ -1,3 +1,13 @@
 import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
 
-export default class ToolsRoute extends Route {}
+export default class ToolsRoute extends Route {
+  @service router;
+  @service currentUser;
+
+  beforeModel() {
+    if (!this.currentUser.adminMember.hasAccess(['isSuperAdmin'])) {
```

Comment:
Remarque, je trouve ça pas hyper facile à comprendre le `!this.currentUser.adminMember.hasAccess(['isSuperAdmin']))` et autre chose, je me demande si un service serait pas à envisager et se mettre du point de vue la route et se "cette route est restreinte aux membre qui sont super admin"

Reply:
- le `!this.currentUser.adminMember.hasAccess(['isSuperAdmin']))` avait été envisagé mercredi quand on était en mob tous ensemble , on avait pas trouvé mieux (voir https://github.com/1024pix/pix/pull/4397) 
- pour le service je ne comprend pas trop ce que tu imagines ?
Est-ce tu peux formuler ta remarque avec des suggestions de code pour mieux comprendre ?

Reply:
Je prends pas longtemps pour regarder et proposer. 
Je pense service car là on a une fonctionnalité qui restreint l'accès à des routes et on délègue ça à une méthode du modèle adminMember, ça pourrait apparaître de manière plus explicite dans le code.

Reply:
je me demande même si ça devrait pas être de la responsabilité du service router (vais voir)

Reply:
ça marche, tant que je n'ai pas d'exemple plus clair je laisse le code tel qu'il est. Si tu as réussis à implémenter quelque chose qui te paraît plus claire pour le service met le en suggestion ou PR à part comme tu veux

Reply:
ça donnerait  dans  admin/app/routes/authenticated/tools.js

```
import Route from '@ember/routing/route';
import { inject as service } from '@ember/service';

export default class ToolsRoute extends Route {
  @service router;
  @service currentUser;
  @service currentDomain;
  @service accessControl;

  beforeModel() {
    this.accessControl.restrictAccessTo(['isSuperAdmin'], 'authenticated');
  }
}
```

et le service 

```
import Service from '@ember/service';
import { inject as service } from '@ember/service';

export default class AccessControlService extends Service {
  @service currentUser;

  restrictAccessTo(roles, redirectionUrl) {
    const currentUserIsAllowed = roles.some((role) => this.currentUser.adminMember[role]);
    if (!currentUserIsAllowed) {
      this.router.transitionTo(redirectionUrl);
    }
  }
}
```

Reply:
on sépare la gestion des acces pour les routes avec un service dédié et la gestion de l'apparition de composant qui reste liée au currentuser role avec le service currentUser
J'ai pas d'avis sur le la logique du hasAccess : je suis pas sure que le M de MVC aient les même responsabilité que le modèle dans la clean archi, là j'ai testé pour mettre casiment toute la logique dans le service

Reply:
J'essaie de faire un commit à part pour proposer ça à la team ! 👀

Reply:
J'ai modifié tout ça (voir dernier commit) !

Reply:
Je trouve ca super clean et donc dans ton commit Anne-G tu pourrais enlever la methode hasAccess je pense

Reply:
j'enlève ça !